### PR TITLE
[FIX] avoid -fsanitizer=undefined false-positive and correct compatibility 2 test.

### PR DESCRIPTION
**Pull request by smehringer**

### CODE REVIEW

File: test/unit/range/view/view_istreambuf_test.cpp [hunk 1]
```diff
@@ -47,17 +47,16 @@ TEST(view_istreambuf, basic)
...
+    EXPECT_TRUE(std::ranges::equal(v3, comp));
 
     // combinability 2
     is.clear();
     is.seekg(0, std::ios::beg);
     auto v4 = view::istreambuf(is) | view::take_until(is_space);
     std::string out2 = v4;
     std::string comp2 = "ACGTATATATAT";
-    EXPECT_TRUE(std::ranges::equal(out, comp));
+    EXPECT_TRUE(std::ranges::equal(out2, comp2));
```
Comment (h-2) [inline test/unit/range/view/view_istreambuf_test.cpp]: haha :clinking_glasses:

Comment (marehr) [reply test/unit/range/view/view_istreambuf_test.cpp]: Is this a different fix?

File: test/unit/range/view/view_istreambuf_test.cpp [hunk 2]
```diff
@@ -47,17 +47,16 @@ TEST(view_istreambuf, basic)
     is.clear();
     is.seekg(0, std::ios::beg);
     auto v3 = view::istreambuf(is) | view::char_to<dna5> | view::complement;
-    std::vector<dna5> out = v3;
     std::vector<dna5> comp{"TGCATATATATANTATATANAATNNNTATATT"_dna5};
-    EXPECT_TRUE(std::ranges::equal(out, comp));
+    EXPECT_TRUE(std::ranges::equal(v3, comp));
```
Comment (marehr) [inline test/unit/range/view/view_istreambuf_test.cpp]: This helped?!

File: test/unit/range/view/view_istreambuf_test.cpp (hunk 1 above)
Comment (h-2) [reply test/unit/range/view/view_istreambuf_test.cpp]: seems like it

Comment (smehringer) [reply test/unit/range/view/view_istreambuf_test.cpp]: Yes :laughing: 
I was really surprised when I made the first change and got an error that out is not known ^^

File: test/unit/range/view/view_istreambuf_test.cpp (hunk 2 above)
Comment (smehringer) [reply test/unit/range/view/view_istreambuf_test.cpp]: Yup... Because now there is no explicit conversion to `std::vector<dna5>` again.

**Pull request merged**